lib/python/qmk/cli/new/keymap.py

```diff
@@ -0,0 +1,65 @@
+"""
+This script automates the copying of the default keymap into your own keymap.
+"""
+import os
+import sys
+
+from milc import cli
+from qmk.path import keymap, check_directories, create_keymap_directory
+
+
+# TODO(orngshdw): add debugging code if needed:
+#   @cli.argument('--debug', dest='debug', action='store_true', help='specify to turn on debugging messages')
+@cli.entrypoint('Create a new keymap.c from the default keyboard of your choosing')
+def main(cli):
+
+    # get keyboard
+    keyboard = input("Keyboard Name: ")
+    # get username
+    username = input("New Username: ")
+
+    # generate keymap paths
+    kb_path = os.path.abspath(keymap(keyboard))
```

Comment:
os.path.abspath() is not necessary here. The current directory for python code is always the top level of `qmk_firmware`.